packages/cloud-functions/src/cico/fetchUserLocationData.ts

```diff
@@ -0,0 +1,60 @@
+import * as functions from 'firebase-functions'
+import { FETCH_TIMEOUT_DURATION, IP_API_KEY } from '../config'
+import { fetchWithTimeout } from './utils'
+
+export interface UserLocationData {
+  countryCodeAlpha2: string | null
+  region: string | null
+  ipAddress: string | null
+}
+
+interface IpAddressData {
+  ip: string
+  version: string
+  city: string
+  region: string
+  region_code: string
+  country: string
+  country_name: string
+  country_code: string
+  country_code_iso3: string
+  country_capital: string
+  country_tld: string
+  continent_code: string
+  in_eu: boolean
+  postal: number
+  latitude: number
+  longitude: number
+  timezone: string
+  utc_offset: number
+  country_calling_code: string
+  currency: string
+  currency_name: string
+  languages: string
+  country_area: number
+  country_population: number
+  asn: string
+  org: string
+}
+
+export const fetchUserLocationData = functions.https.onRequest(async (req, res) => {
+  const ipAddress =
+    req.headers['x-appengine-user-ip'] ||
+    req.headers['x-forwarded-for'] ||
+    req.connection.remoteAddress
+
+  const response = await fetchWithTimeout(
+    `http://api.ipapi.com/api/${ipAddress}?access_key=${IP_API_KEY}`,
+    null,
+    FETCH_TIMEOUT_DURATION
```

Comment:
Since we're getting this from the config anyways, should this `FETCH_TIMEOUT_DURATION` be the default and the timeout an optional parameter?

Reply:
Good suggestion, thanks. I made some changes to this function in [my other PR ](https://github.com/celo-org/wallet/pull/384) so I will incorporate the default value there to minimize merge conflicts